### PR TITLE
Optimize docker + Add environmental variable

### PR DESCRIPTION
**Pull request by connorv001**
- Enable prefixes based on environment variables
- Add Docker-Compose and Docker build instructions
- Update usage example with Docker instructions
- Change base image from `node` to `node:bullseye-slim`
- Install `chromium`

[src/index.ts]
- Enabled prefixes based on environment variables
- Set default values for GPT and Dalle prefixes
- Changed the Whatsapp Client initialization [docker-compose.yml]
- Add docker-compose.yml file
- Set environment variables for OPENAI_API_KEY, PREFIX_ENABLED, GPT_PREFIX, and DALLE_PREFIX [Dockerfile]
- Change base image from `node` to `node:bullseye-slim`
- Set `WORKDIR` to `/app/`
- Add environment variables `OPENAI_API_KEY`, `PREFIX_ENABLED`, `GPT_PREFIX` and `DALLE_PREFIX`
- Move `WORKDIR` from `/tmp/app/` to `/app/`
- Install `chromium [README.md]
- Add Docker-Compose instructions
- Add Docker build instructions
- Update usage example

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: '3'
+services:
+  whatsapp-chatgpt:
+    image: ghcr.io/askrella/whatsapp-chatgpt:latest
```
Comment (EzxD) [inline docker-compose.yml:4]: i guess we need a pipeline to made this work

Comment (connorv001) [reply docker-compose.yml:4]: Oh yes yes.

Comment (EzxD) [reply docker-compose.yml:4]: would you be so kind and add a pipeline too?

Comment (connorv001) [reply docker-compose.yml:4]: > would you be so kind and add a pipeline too?

Of course adding

Comment (connorv001) [reply docker-compose.yml:4]: @EzxD  added kind sir

**Comment by connorv001** [comment]
@navopw  Any problem with this?

**Comment by steve-hb** [comment]
The pipeline seems good, how big is the container resulting from the Dockerfile? Maybe we can optimize this part a little bit and then I'd approve this :)

**Comment by connorv001** [comment]
> The pipeline seems good, how big is the container resulting from the Dockerfile? Maybe we can optimize this part a little bit and then I'd approve this :)

As of the last commit, it's 1.52GB.. but still smaller than the original Dockerfile. I aim to optimize

**Review by steve-hb — APPROVED** [review]

**Comment by connorv001** [comment]
@steve-hb  unfortunately this is the working most stable size right now. I tried it with couple of other things, because of chromium it takes up atleast 900 MB
But this is the most stable version as far as docker is cocnerned.

**Comment by steve-hb** [comment]
Thanks, approved by Paul and me. @navopw still needs to review it, but we recommend merging :)

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,19 @@
-FROM node
+FROM node:bullseye-slim
+
+WORKDIR /app/
+
+ENV OPENAI_API_KEY ""
+ENV PREFIX_ENABLED ""
+ENV GPT_PREFIX ""
+ENV DALLE_PREFIX ""
+ENV REPLY_SELF_ENABLED "true"
```
Comment (navopw) [inline Dockerfile:9]: Remove
- GPT_PREFIX
- DALLE_PREFIX
- REPLY_SELF_ENABLED

File: README.md [hunk 3]
```diff
@@ -23,6 +23,9 @@ Whatsapp bot that uses OpenAI's GPT & DALLE to respond to user inputs.
 ```
 OPENAI_API_KEY=put_your_key_here
 PREFIX_ENABLED=false
+GPT_PREFIX=!gpt
```
Comment (navopw) [inline README.md:26]: Remove

GPT_PREFIX
DALLE_PREFIX
REPLY_SELF_ENABLED

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+version: '3'
+services:
+  whatsapp-chatgpt:
+    image: ghcr.io/askrella/whatsapp-chatgpt:main
+    environment:
+      OPENAI_API_KEY: ""
+      PREFIX_ENABLED: ""
+      GPT_PREFIX: ""
```
Comment (navopw) [inline docker-compose.yml:8]: Remove

GPT_PREFIX
DALLE_PREFIX
REPLY_SELF_ENABLED

File: src/index.ts [hunk 5]
```diff
@@ -13,10 +13,12 @@ import { handleMessageDALLE } from "./dalle";
 const statusBroadcast = "status@broadcast";
 
 // Prefixes
+
```
Comment (navopw) [inline src/index.ts:16]: Remove empty line

File: src/index.ts [hunk 6]
```diff
@@ -13,10 +13,12 @@ import { handleMessageDALLE } from "./dalle";
...
 const prefixEnabled = process.env.PREFIX_ENABLED == "true";
 const gptPrefix = "!gpt";
 const dallePrefix = "!dalle";
 
+
```
Comment (navopw) [inline src/index.ts:21]: Remove empty line

**Pull request merged**